src/hera/retry_strategy.py

```diff
@@ -0,0 +1,53 @@
+from dataclasses import dataclass
+from typing import Dict, Optional
+
+from argo_workflows.models import (
+    IoArgoprojWorkflowV1alpha1Backoff,
+    IoArgoprojWorkflowV1alpha1RetryAffinity,
+    IoArgoprojWorkflowV1alpha1RetryStrategy,
+)
+
+from hera.retry_policy import RetryPolicy
+
+
+@dataclass
+class RetryStrategy:
+    """Retry holds the duration values for retrying tasks.
+
+    Attributes
+    ----------
+    affinity: Optional[Dict] = None
+        Affinity prevents running workflow's step on the same host.
+        See: https://argoproj.github.io/argo-workflows/fields/#retryaffinity
+    backoff: Optional[Dict] = None
```

Comment:
What do you think about a custom `Backoff` object for this? 🤔 Seems like it's pretty simple, with 3 fields only. It would save users from having to reference the Argo docs

Reply:
We could, but then I think we should add it everywhere we're currently using dicts as valid inputs.

Reply:
Maybe in a future argo version they'll add some more parameters. If we support dict we don't need to update anything.

Reply:
I see your perspective. Perhaps we can add it in the future! Also, I want to mention that these things can be introduced gradually. I appreciate that we should add it everywhere, such as to `ValueFrom`, but even that can be introduced one by one! 

Thank you for the discussion 🙂